api/v1alpha4/conversion_test.go

```diff
@@ -58,6 +60,21 @@ func TestFuzzyConversion(t *testing.T) {
 	}))
 }
 
+func MachineSpecFuzzFunc(_ runtimeserializer.CodecFactory) []interface{} {
+	return []interface{}{
+		MachineSpecFuzzer,
+	}
+}
+
+func MachineSpecFuzzer(in *MachineSpec, c fuzz.Continue) {
+	c.FuzzNoCustom(in)
+
+	// Version field has been converted from *string to string in v1beta1,
+	// so we're forcing valid string values to avoid round trip errors.
+	versionString := c.RandString()
+	in.Version = &versionString
```

Comment:
It seems here the problem only appears when in.Version has been generated to be a `nil` value. Could we only force a string here `if in.Version == nil {` ?

Reply:
Done!